src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1382,21 +1352,63 @@ void walkInnerSymbols(ObjectType type, String innerNamespace) {
           privateEnums.add(innerNamespace + '.' + propName);
         }
       }
+      boolean nonEmpty = false;
 
       for (String propName : getSortedPublicPropertyNames(type)) {
         String qualifiedName = innerNamespace + '.' + propName;
         if (provides.contains(qualifiedName)) continue;
         JSType pType = type.getPropertyType(propName);
-
         if (pType.isEnumType()) {
+          if (!nonEmpty) {
+            emitNamespaceBegin(innerNamespace);
```

Comment:
I hate this pattern, but its the only foolproof way to guarantee that no empty namespaces get emitted. Any ideas how to clean it up?